fe/fe-core/src/main/java/com/starrocks/privilege/PrivilegeManager.java

```diff
@@ -406,8 +500,16 @@ protected PrivilegeCollection mergePrivilegeCollection(ConnectContext context) t
             if (!userToPrivilegeCollection.containsKey(userIdentity)) {
                 throw new PrivilegeException("cannot find " + userIdentity.toString());
             }
-            // TODO merge role privilege
-            return userToPrivilegeCollection.get(userIdentity);
+            PrivilegeCollection collection = new PrivilegeCollection();
+            UserPrivilegeCollection userCollection = userToPrivilegeCollection.get(userIdentity);
+            collection.merge(userCollection);
```

Comment:
Do we need to optimize here? Because every time we need to check action, we will do a merge of all priviledges belonged to the user. How about we don't do the merge, just check directly on the user priviledges and roles?

Reply:
I have considered caching privilege collections right after login as a further improvement. This is just step one. If caching is implemented, it would be much faster than checking by roles and user in turns.

Reply:
Caching is better, i agree.